src/worker.cpp

```diff
@@ -239,37 +239,29 @@ std::string Worker::getHWVersion(const types::IPZVpdMap& parsedVpd) const
     return hwString.str();
 }
 
-void Worker::getVpdDataInVector(const std::string& vpdFilePath,
-                                types::BinaryVector& vpdVector,
-                                size_t& vpdStartOffset)
+void Worker::fillVPDMap(const std::string& vpdFilePath,
+                        types::VPDMapVariant& vpdMap)
 {
-    std::fstream vpdFileStream;
-    vpdFileStream.exceptions(std::ifstream::badbit | std::ifstream::failbit);
+    logging::logMessage(std::string("Parsing file = ") + vpdFilePath);
 
-    try
+    if (vpdFilePath.empty())
     {
-        vpdFileStream.open(vpdFilePath,
-                           std::ios::in | std::ios::out | std::ios::binary);
-        auto vpdSizeToRead = std::min(std::filesystem::file_size(vpdFilePath),
-                                      static_cast<uintmax_t>(65504));
-        vpdVector.resize(vpdSizeToRead);
-
-        vpdFileStream.seekg(vpdStartOffset, std::ios_base::beg);
-        vpdFileStream.read(reinterpret_cast<char*>(&vpdVector[0]),
-                           vpdSizeToRead);
-
-        vpdVector.resize(vpdFileStream.gcount());
-        vpdFileStream.clear(std::ios_base::eofbit);
+        throw std::runtime_error("Invalid file path passed to fillVPDMap API.");
     }
-    catch (const std::ifstream::failure& fail)
+
+    if (!std::filesystem::exists(vpdFilePath))
     {
-        std::cerr << "Exception in file handling [" << vpdFilePath
-                  << "] error : " << fail.what();
-        std::cerr << "Stream file size = " << vpdFileStream.gcount()
-                  << std::endl;
-        throw;
+        throw std::runtime_error("Can't Find physical file");
```

Comment:
can you please add filepath in log message

Reply:
since vpd is collected parallely, we may end up getting the print at line 245 for two or more eeprom paths. in that case `Can't Find physical file` will be of no use without the file name given.

Reply:
Not yet, it is not being collected in parallel, we might put this in api as critical section when we implement that part. Can take a call at that time based on implementation.